chorus_lib/src/transport/local.rs

```diff
@@ -64,6 +66,58 @@ impl<L: HList> LocalTransportChannel<L> {
     }
 }
 
+/// A builder for `LocalTransportChannel`.
+///
+/// Use this builder to create a `LocalTransportChannel` instance.
+///
+/// # Examples
+///
+/// ```
+/// use chorus_lib::core::{LocationSet, ChoreographyLocation};
+/// use chorus_lib::transport::local::{LocalTransportChannelBuilder};
+///
+/// #[derive(ChoreographyLocation)]
+/// struct Alice;
+///
+/// #[derive(ChoreographyLocation)]
+/// struct Bob;
+///
+/// let transport_channel = LocalTransportChannelBuilder::new()
+///     .with(Alice)
+///     .with(Bob)
+///     .build();
+/// ```
+pub struct LocalTransportChannelBuilder<L: HList> {
+    location_set: PhantomData<L>,
+}
+
+impl LocalTransportChannelBuilder<LocationSet!()> {
+    /// Creates a new `LocalTransportChannelBuilder` instance
+    pub fn new() -> Self {
+        Self {
+            location_set: PhantomData,
+        }
+    }
+}
+
+impl<L: HList> LocalTransportChannelBuilder<L> {
+    /// Adds a new location to the set of locations in the `LocalTransportChannel`.
+    pub fn with<NewLocation: ChoreographyLocation>(
+        &self,
+        location: NewLocation,
+    ) -> LocalTransportChannelBuilder<HCons<NewLocation, L>> {
+        _ = location;
```

Comment:
Do you think it's better to do this, or have `_location = NewLocation` on line 107? (We have the same thing [here](https://github.com/lsd-ucsc/ChoRus/blob/0c419a582da7716573ca96042f1397fc09369464/chorus_lib/src/transport.rs#L71)

Reply:
This is intentional. I don't want to show underscores on editors.
![Screenshot 2023-10-02 at 12 29 40](https://github.com/lsd-ucsc/ChoRus/assets/10496155/0bb7b63c-2feb-4bb4-9eb9-c2057d17a80b)

Reply:
Fixed several more places 
d4f96a4ab1c168e904790462a5679c67da22385d